### DISCUSSION TITLE
Did we lose mdx -> markdown preview for good?

### DISCUSSION THREAD

**Discussion by djw-m** [community/community#60747] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

After the last UI refresh, I noticed that wheras previously the preview button on an mdx file would show you a rendering of the markdown, it has now switched to showing you a view of the diffs. Two things - this is more a view of the changes you are making, and there's no way to quickly see the results of the changes to the markdown content. If there's a way to revert this behavior I would love to know.

(and, for the future, github, could we have a choice of how we preview changes - a diff view and a markdown preview on mdx would be splendid.)

**Comment by sfirke** [comment]
I was trying to figure out how to see a rendering of a .mdx markdown file I was editing on GitHub - I couldn't find a way, but found this.  Would love to see .mdx rendered like .md.

**Comment by evichtran** [comment]
Markdown is also good!

**Comment by jeska** [comment]
Adding a +1 to this as a concern. Our team heavily leverages `.mdx` files for rendering, and losing this feature is a loss for us. We are able to render in IDEs such as `vscode`, but being able to quickly check in the Github UI (whether looking at an existing file or update in a PR) was always useful in a pinch.

**Comment by polly-bold** [comment]
Our team also leverages MDX quite heavily and would appreciate a markdown preview of .mdx files.

**Comment by jamietanna** [comment]
It seems to be back :clap:

**Comment by djw-m** [comment]
It is and the qol is way up for working on MDX files

**Comment by AMorgaut** [comment]
I started recently to use mdx, mostly because of storybook, so I would be interested too

BUT I'm conflicted

MDX means code execution and running arbitrary community provided code is giving all the weapons to hackers 

As an example, such script could be using github connected visitors session to make them merge code in their own repositories without their will with benefits of this request being made from the github origin (so no CORS or same-origin like protections).

I'm surprised to read it even worked before. It would need to run in a sandbox with at least a different origin
